pkgs/development/python-modules/boto3/default.nix

```diff
@@ -1,50 +1,57 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , botocore
 , jmespath
 , s3transfer
-, futures ? null
-, docutils
-, nose
-, mock
-, isPy3k
+, setuptools
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "boto3";
-  version = "1.24.75"; # N.B: if you change this, change botocore and awscli to a matching version
+  version = "1.26.32"; # N.B: if you change this, change botocore and awscli to a matching version
+  format = "pyproject";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-e/oiQt+bH6c+F/TX959rIlYPVdXaLifleOWF725LFKc=";
+  src = fetchFromGitHub {
+    owner = "boto";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-XHySvLkOt0M5u4wVP1FuTcfqgNsfRo1WHWr75Qz+m+g=";
   };
 
-  propagatedBuildInputs = [ botocore jmespath s3transfer ] ++ lib.optionals (!isPy3k) [ futures ];
-  checkInputs = [ docutils nose mock ];
+  propagatedBuildInputs = [
+    botocore
+    jmespath
+    s3transfer
+    setuptools
+  ];
 
-  checkPhase = ''
-    runHook preCheck
-    # This method is not in mock. It might have appeared in some versions.
-    sed -i 's/action.assert_called_once()/self.assertEqual(action.call_count, 1)/' \
-      tests/unit/resources/test_factory.py
-    nosetests -d tests/unit --verbose
-    runHook postCheck
-  '';
+  doCheck = true;
 
-  # Network access
-  doCheck = false;
+  checkInputs = [
+    pytestCheckHook
+  ];
 
-  pythonImportsCheck = [ "boto3" ];
+  pythonImportsCheck = [
+    "boto3"
+  ];
 
-  meta = {
+  disabledTestPaths = [
+    # Integration tests require networking
+    "tests/integration"
```

Comment:
```suggestion
    "tests/integration"
    "tests/test_kinesis/test_kinesis.py"
    "tests/test_kinesis/test_kinesis_stream_consumers.py"
```

Reply:
Added to disabledTestPaths of pkgs/development/python-modules/moto/default.nix